womtool/src/main/scala/cromwell/pipeline/womtool/WomTool.scala

```diff
@@ -0,0 +1,78 @@
+package cromwell.pipeline.womtool
+
+import java.nio.file.{ Files, Paths }
+
+import cats.data.NonEmptyList
+import com.typesafe.config.ConfigFactory
+import common.Checked
+import cromwell.core.path.{ DefaultPath, DefaultPathBuilder, Path }
+import cromwell.languages.LanguageFactory
+import cromwell.languages.util.ImportResolver.ImportResolver
+import languages.cwl.CwlV1_0LanguageFactory
+import languages.wdl.biscayne.WdlBiscayneLanguageFactory
+import languages.wdl.draft2.WdlDraft2LanguageFactory
+import languages.wdl.draft3.WdlDraft3LanguageFactory
+import wom.executable.WomBundle
+import womtool.WomtoolMain.{ BadUsageTermination, SuccessfulTermination, Termination, UnsuccessfulTermination }
+import womtool.inputs.Inputs
+
+import scala.util.Try
+
+class WomTool extends WomToolAPI {
+
+  def validate(
+    content: String,
+    importResolvers: List[ImportResolver]
+  ): Either[NonEmptyList[String], (WomBundle, LanguageFactory)] = {
+
+    val languageFactory: LanguageFactory =
+      List(
+        new WdlDraft3LanguageFactory(ConfigFactory.empty()),
+        new WdlBiscayneLanguageFactory(ConfigFactory.empty()),
+        new CwlV1_0LanguageFactory(ConfigFactory.empty())
+      ).find(_.looksParsable(content)).getOrElse(new WdlDraft2LanguageFactory(ConfigFactory.empty()))
```

Comment:
I suggest we extract the list of factories to the object field, and in the method just search for matching one:
```scala
class WomTool extends WomToolAPI {
  private val defaultLanguageFactory = new WdlDraft2LanguageFactory(ConfigFactory.empty())
  private val languageFactories = List(
    new WdlDraft3LanguageFactory(ConfigFactory.empty()),
    new WdlBiscayneLanguageFactory(ConfigFactory.empty()),
    new CwlV1_0LanguageFactory(ConfigFactory.empty())
  )
  /* ... */
}
```
```suggestion
    val languageFactory = languageFactories.find(_.looksParsable(content)).getOrElse(defaultLanguageFactory)
```

Reply:
Have done!